components/shooter.py

```diff
@@ -1,34 +1,73 @@
-GEAR_RATIO: float  # whatever this is
+from math import radians, tau
 
+from magicbot import tunable
+from rev import CANSparkMax
+from wpilib import DigitalInput
 
-class Shooter:
-    # top_flywheel: whatever motor type this is
-    # bottom flywheel: whatever motor type this is
-    # neck_motor: whatever motor this is <- this refers to the back motor that pulls the cube away from the motors
+from ids import DioChannels, SparkMaxIds
+
+GEAR_RATIO: float = 1 / 1
+FLYWHEEL_SPEED_ERROR_TOLERANCE: float = radians(1)
 
-    # loaded_switch Digital I/O <- in my head this has something to confirm when the shooter is correctly loaded
 
+class Shooter:
     def __init__(self) -> None:
         # create instances of hardware handles
+        self.top_flywheel = CANSparkMax(
+            SparkMaxIds.top_flywheel, CANSparkMax.MotorType.kBrushless
+        )
+        self.bottom_flywheel = CANSparkMax(
+            SparkMaxIds.bottom_flywheel, CANSparkMax.MotorType.kBrushless
+        )
+        self.back_motor = CANSparkMax(
+            SparkMaxIds.back_motor, CANSparkMax.MotorType.kBrushless
+        )
+
+        self.top_flywheel_encoder = self.top_flywheel.getEncoder()
+        self.bottom_flywheel_encoder = self.bottom_flywheel.getEncoder()
+        self.top_flywheel_encoder.setVelocityConversionFactor(GEAR_RATIO * tau / 60)
+        self.bottom_flywheel_encoder.setVelocityConversionFactor(GEAR_RATIO * tau / 60)
+
+        self.loaded_switch = DigitalInput(DioChannels.loaded_shooter)
 
         # initialise motor speed
+        self.top_flywheel_speed = tunable(0.0)
+        self.bottom_flywheel_speed = tunable(0.0)
```

Comment:
As the type checker has caught, this is incorrect usage of `tunable`. They need to be assigned to a class variable.

Docs:

- https://robotpy.readthedocs.io/en/stable/frameworks/magicbot.html#dashboard-coprocessor-communications
- https://robotpy.readthedocs.io/projects/utilities/en/stable/magicbot.html#magicbot.magic_tunable.tunable

```suggestion
        top_flywheel_speed = tunable(0.0)
        bottom_flywheel_speed = tunable(0.0)
```

Alternatively, don't use `tunable` for now:

```suggestion
        self.top_flywheel_speed = 0.0
        self.bottom_flywheel_speed = 0.0
```